File: bench/bench_ocaml/test/test_core.ml
module Math = Math_lib.Math
module Test = Math_lib.Common.Test

let _test_isqrt ?(log = false) () =
  let () = assert (Math.isqrt (Z.of_int 4) = Z.of_int 2) in
  let () = assert (Math.isqrt (Z.of_int 8) = Z.of_int 2) in
  let () = assert (Math.isqrt (Z.of_int 9) = Z.of_int 3) in
  let () = assert (Math.isqrt (Z.of_int 15) = Z.of_int 3) in
  let () = assert (Math.isqrt (Z.of_int 16) = Z.of_int 4) in
  let () = assert (Math.isqrt (Z.of_int 17) = Z.of_int 4) in

  if log then Test.log "Test 'isqrt' finished"

let _test_factorial ?(log = false) () =
  let () = assert (Math.factorial (Z.of_int 0) = Z.of_int 1) in
  let () = assert (Math.factorial (Z.of_int 1) = Z.of_int 1) in
  let () = assert (Math.factorial (Z.of_int 2) = Z.of_int 2) in
  let () = assert (Math.factorial (Z.of_int 3) = Z.of_int 6) in
  let () = assert (Math.factorial (Z.of_int 4) = Z.of_int 24) in
  let () = assert (Math.factorial (Z.of_int 5) = Z.of_int 120) in
  let () = assert (Math.factorial (Z.of_int 6) = Z.of_int 720) in

  if log then Test.log "Test 'factorial' finished"

let all ?log () =
  _test_isqrt ?log ();
  _test_factorial ?log ()

File: bench/bench_kotlin/src/test/kotlin/math/TestCore.kt
package math

import math.lib.Math
import java.math.BigInteger
import kotlin.test.Test
import kotlin.test.assertEquals

internal class TestCore {

    @Test
    fun testISqrt() {
        assertEquals(Math.iSqrt(BigInteger.valueOf(4)), BigInteger.TWO)
        assertEquals(Math.iSqrt(BigInteger.valueOf(8)), BigInteger.TWO)
        assertEquals(Math.iSqrt(BigInteger.valueOf(9)), BigInteger.valueOf(3))
        assertEquals(Math.iSqrt(BigInteger.valueOf(15)), BigInteger.valueOf(3))
        assertEquals(Math.iSqrt(BigInteger.valueOf(16)), BigInteger.valueOf(4))
    }

    @Test
    fun testFactorial() {
        assertEquals(Math.factorial(BigInteger.ZERO), BigInteger.ONE)
        assertEquals(Math.factorial(BigInteger.ONE), BigInteger.ONE)
        assertEquals(Math.factorial(BigInteger.TWO), BigInteger.TWO)
        assertEquals(Math.factorial(BigInteger.valueOf(3)), BigInteger.valueOf(6))
        assertEquals(Math.factorial(BigInteger.valueOf(4)), BigInteger.valueOf(24))
        assertEquals(Math.factorial(BigInteger.valueOf(5)), BigInteger.valueOf(120))
        assertEquals(Math.factorial(BigInteger.valueOf(6)), BigInteger.valueOf(720))
    }
}